exercises/practice/bob/.approaches/when-expression/content.md

```diff
@@ -0,0 +1,60 @@
+# `when` expressions
+
+```kotlin
+object Bob {
+    fun hey(statement: String): String =
+        when {
+            statement.isQuestion() && statement.isYelling() -> "Calm down, I know what I'm doing!"
+            statement.isQuestion() -> "Sure."
+            statement.isYelling() -> "Whoa, chill out!"
+            statement.isSilence() -> "Fine. Be that way!"
+            else -> "Whatever."
+        }
+
+    private fun String.isSilence(): Boolean = this.isBlank()
+    private fun String.isQuestion(): Boolean = this.trim().endsWith('?')
```

Comment:
The return types could be omitted (feel free to ignore this suggestion)

```suggestion
    private fun String.isSilence() = this.isBlank()
    private fun String.isQuestion() = this.trim().endsWith('?')
```

Reply:
I usually really like explicit return types because I think it makes the intent and outcome of a function easier to read.

But I know that you are able to omit them in Kotlin if you want to. 
But I usually don't omit them.

Reply:
Okay, that's perfectly fine :)